tools/amd_build/pyHIPIFY/hipify-python.py

```diff
@@ -0,0 +1,966 @@
+    """Disables calls to an unsupported HIP function"""
+    # Prepare output string
+    output_string = input_string
+
+    # Find all calls to the function
+    calls = re.finditer(r"\b%s\b" % function, input_string)
+
+    # Do replacements
+    for call in calls:
+        start = call.start()
+        end = call.end()
+
+        pos = end
+        started_arguments = False
+        bracket_count = 0
+        while pos < len(input_string):
+            if input_string[pos] == "(":
+                if started_arguments is False:
+                    started_arguments = True
+                    bracket_count = 1
+                else:
+                    bracket_count += 1
+            elif input_string[pos] == ")" and started_arguments:
+                bracket_count -= 1
+
+            if bracket_count == 0 and started_arguments:
+                # Finished!
+                break
+            pos += 1
+
+        function_call = input_string[start:pos + 1]
+        output_string = output_string.replace(function_call, replacement)
+
+    return output_string
+
+
+def disable_module(input_file):
+    """Disable a module entirely except for header includes."""
+    with openf(input_file, "r+") as f:
+        txt = f.read()
+        last = list(re.finditer(r"#include .*\n", txt))[-1]
+        end = last.end()
+
+        disabled = "%s#if !defined(__HIP_PLATFORM_HCC__)\n%s\n#endif" % (txt[0:end], txt[end:])
+
+        f.seek(0)
+        f.write(disabled)
+        f.truncate()
+
+
+def extract_arguments(start, string):
+    """ Return the list of arguments in the upcoming function parameter closure"""
+    arguments = []
+    closures = {
+        "<": 0,
+        "(": 0
+    }
+    current_position = start
+    argument_start_pos = current_position + 1
+
+    # Search for final parenthesis
+    while current_position < len(string):
+        if string[current_position] == "(":
+            closures["("] += 1
+        elif string[current_position] == ")":
+            closures["("] -= 1
+        elif string[current_position] == "<":
+            closures["<"] += 1
+        elif string[current_position] == ">" and string[current_position - 1] != "-":
+            closures["<"] -= 1
+
+        # Finished all arguments
+        if closures["("] == 0 and closures["<"] == 0:
+            # Add final argument
+            arguments.append({"start": argument_start_pos, "end": current_position})
+            break
+
+        # Finished current argument
+        if closures["("] == 1 and closures["<"] == 0 and string[current_position] == ",":
+            arguments.append({"start": argument_start_pos, "end": current_position})
+            argument_start_pos = current_position + 1
+
+        current_position += 1
+
+    return arguments
+
+
+def add_static_casts(directory, extensions, KernelTemplateParams):
+    """Added necessary static casts to kernel launches due to issue in HIP"""
```

Comment:
A great way to make this documentation more clear is to give some example code before and after this transformation.

Also, if you cite an issue in HIP, best to cite the actual issue (e.g. a URL), so that it's easy to see if the problem in HIP has been fixed or not.

Reply:
It's actually not an *issue* in HIP. Rather, it's to ensure that the type of the kernel arguments matches the type specified in the kernel definition. I have updated the comments and added a before-and-after example in commit commit acfacbc as part of PR https://github.com/ROCmSoftwarePlatform/pytorch/pull/20